pkg/providers/docker/docker.go

```diff
@@ -212,6 +213,8 @@ func buildTemplateMapCP(clusterSpec *cluster.Spec) map[string]interface{} {
 		"auditPolicy":                common.GetAuditPolicy(),
 		"podCidrs":                   clusterSpec.Spec.ClusterNetwork.Pods.CidrBlocks,
 		"serviceCidrs":               clusterSpec.Spec.ClusterNetwork.Services.CidrBlocks,
+		"haproxyImageRepository":     strings.ReplaceAll(bundle.Haproxy.Image.Image(), "/haproxy", ""),
```

Comment:
CAPD expects only the partial repository URI prefix here, as it appends `haproxy` to the prefix to get the final URI. By default this field is set to `kindest`, so the full URI becomes `kindest/haproxy`. We override it by setting it to `public.ecr.aws/<registry alias>/kubernetes-sigs/kind`, to form the final URI `public.ecr.aws/<registry alias>/kubernetes-sigs/kind/haproxy`.

Reply:
I tried using filepath.Dir(), but it returned a `.` for empty string input, which isn't ideal. If using that and having a check in the template to replace `.` with empty string is more preferrable, I can change to using that.

Reply:
instead of using replaceAll, can we possibly check if the image follows the patten (regex) and then remove the last `/haproxy` part?

Reply:
Updated.